qiskit/algorithms/gradients/spsa_sampler_gradient.py

```diff
@@ -0,0 +1,125 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Gradient of Sampler with Finite difference method."""
+
+from __future__ import annotations
+
+from typing import Sequence
+
+import numpy as np
+
+from qiskit.circuit import Parameter, QuantumCircuit
+from qiskit.exceptions import QiskitError
+from qiskit.primitives import BaseSampler
+from qiskit.providers import JobStatus
+
+from .base_sampler_gradient import BaseSamplerGradient
+from .sampler_gradient_result import SamplerGradientResult
+
+
+class SPSASamplerGradient(BaseSamplerGradient):
+    """
+    Compute the gradients of the sampling probability by the Simultaneous Perturbation Stochastic
+    Approximation (SPSA).
+    """
+
+    def __init__(
+        self,
+        sampler: BaseSampler,
+        epsilon: float,
+        batch_size: int = 1,
+        seed: int | None = None,
+        **run_options,
+    ):
+        """
+        Args:
+            sampler: The sampler used to compute the gradients.
+            epsilon: The offset size for the SPSA gradients.
+            batch_size: number of gradients to average.
+            seed: The seed for a random perturbation vector.
+            run_options: Backend runtime options used for circuit execution. The order of priority is:
+                run_options in `run` method > gradient's default run_options > primitive's default
+                setting. Higher priority setting overrides lower priority setting.
+
+        Raises:
+            ValueError: If ``epsilon`` is not positive.
+        """
+        if epsilon <= 0:
+            raise ValueError(f"epsilon ({epsilon}) should be positive.")
+        self._batch_size = batch_size
+        self._epsilon = epsilon
+        self._seed = np.random.default_rng(seed) if seed else np.random.default_rng()
```

Comment:
> self._seed = np.random.default_rng(seed) if seed else np.random.default_rng()

default_rng() returns a generator so I would think passing into an var named _seed seems odd self._rng perhaps or self._random. So

self._random = np.random.default_rng(seed)

As you can pass None into default_rng and it will randomly seed it so the if seed part is not needed